### PR TITLE
preparing for v1.8

### PR DESCRIPTION
**Pull request by samuelcolvin**
:tada: :rocket:

### CODE REVIEW

**Comment by samuelcolvin** [comment]
Humm, I hadn't noticed that the coverage has fallen, I need to fix that before release.

**Comment by cybojenix** [comment]
Just run our tests targeting the master branch, and no regressions.
We don't use all the features, but it's a good sign that even our use of some private APIs works correctly (mainly on `validate_arguments`).

Thanks!

**Comment by samuelcolvin** [comment]
> Just run our tests targeting the master branch, and no regressions.
> We don't use all the features, but it's a good sign that even our use of some private APIs works correctly (mainly on `validate_arguments`).

Thanks for checking! So useful when a few people run it against their tests before rather than after release.

File: HISTORY.md [hunk 1]
```diff
@@ -1,3 +1,102 @@
...
+* fix: update all modified field values in `root_validator` when `validate_assignment` is on, #2116 by @PrettyWood
+* Allow pickling of `pydantic.dataclasses.dataclass` dynamically created from a built-in `dataclasses.dataclass`, #2111 by @aimestereo
+* Fix a regression where Enum fields would not propagate keyword arguments to the schema, #2109 by @bm424
+* Add the ability to customize settings sources (add / disable / change priority order), #2107 by @kozlek
+* Fix mypy complaints about most custom _pydantic_ types, #2098 by @PrettyWood
+* Add a [Hypothesis](https://hypothesis.readthedocs.io/) plugin for easier [property-based testing](https://increment.com/testing/in-praise-of-property-based-testing/) with Pydantic's custom types - [usage details here](https://pydantic-docs.helpmanual.io/hypothesis_plugin/), #2097 by @Zac-HD
+* add validator for `None`, `NoneType` or `Literal[None]`, #2095 by @PrettyWood
+* Handle properly fields of type `Callable` with a default value, #2094 by @PrettyWood
+* Ignore `__doc__` as private attribute when `Config.underscore_attrs_are_private` is set, #2090 by @PrettyWood
```
Comment (PrettyWood) [inline HISTORY.md:79]:
```suggestion
```
I think some stuff comes from 1.7.3

Comment (samuelcolvin) [reply HISTORY.md:79]: good spot, will remove.

Comment (samuelcolvin) [reply HISTORY.md]: done.

**Pull request merged**